tests/callbacks/test_lr_monitor.py

```diff
@@ -150,8 +182,22 @@ def test_lr_monitor_param_groups(tmpdir):
     """ Test that learning rates are extracted and logged for single lr scheduler. """
     tutils.reset_seed()
 
-    model = EvalModelTemplate()
-    model.configure_optimizers = model.configure_optimizers__param_groups
+    class LessBoringModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.learning_rate = 0.1
+
+        def configure_optimizers(self):
+            param_groups = [
+                {'params': list(self.parameters())[:2], 'lr': self.learning_rate * 0.1},
+                {'params': list(self.parameters())[2:], 'lr': self.learning_rate}
+            ]
+
+            optimizer = optim.Adam(param_groups)
+            lr_scheduler = optim.lr_scheduler.StepLR(optimizer, 1, gamma=0.1)
+            return [optimizer], [lr_scheduler]
```

Comment:
BoringModel won't be having any variations for it's methods? Is it so Boring? :joy:

Reply:
We can overwrite / add new methods to make less boring 🤣, but only in the tests.

Reply:
Yeah but it's like writing the same code again and again if same variation of some method is required at different places.